### ISSUE TITLE
NET-3004 Fix S2696 FP: New extension method format triggers FP when accessing static fields

### ISSUE THREAD

**Issue by dougwaldron** [SonarSource/sonar-dotnet#9783]
### Description

The new C# 14 extension method syntax introduces a false positive for [S2696](https://rules.sonarsource.com/csharp/RSPEC-2696/) if the method updates a `static` field. This warning was not raised with the old syntax.

### Reproducer

```csharp
public static class ProvidersExtensions
{
    private static IEnumerable<string>? _providers;

    extension(IConfiguration configuration)
    {
        public IEnumerable<string> GetProvidersNewFormat()
        {
            // The following line triggers the S2696 warning.
            _providers ??= configuration.GetSection("Providers").Get<string[]>() ?? [];
            return _providers;
        }
    }

    public static IEnumerable<string> GetProvidersOldFormat(this IConfiguration configuration)
    {
        // The following line does not trigger the warning.
        _providers ??= configuration.GetSection("Providers").Get<string[]>() ?? [];
        return _providers;
    }
}
```

### Product and Version

SonarAnalyzer.CSharp 10.18.0.131500

**Comment by Corniel** [comment]
The claimed they fixed this: https://community.sonarsource.com/t/fp-s2325-does-report-on-c-14-extensions/151936

**Comment by alex-meseldzija-sonarsource** [comment]
Hi @dougwaldron,
Thanks for the report, this is an FP.

While it is similar to the S2325 FP, unfortunately they do not share an implementation so that fix is specific to that rule.

**Comment by Tim-Pohlmann** [comment]
Hi @dougwaldron, this False Positive has been fixed and is now available on SonarQube Cloud and will soon come to SonarQube Server 2026.2 and SonarQube Community Build 26.3.
